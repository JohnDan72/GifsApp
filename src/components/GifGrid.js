import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Col, Divider, List, Row } from 'rsuite';
import GifGridItem from './GifGridItem';

const propTypes = {
    category: PropTypes.string.isRequired
};

const defaultProps = {};

/**
 * 
 */

const GifGrid = ({ category }) => {
    const [count, setCount] = useState(0)
    const [gifs, setgifs] = useState([])
    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?api_key=0bcW1FR66wVpVSj3v8logtrzaY6OIjvO&q=matrix&limit=25&offset=0&rating=g&lang=en`;
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs_data = data.map(img => ({ id: img.id, title: img.title, url: img.images?.downsized_medium.url }))
        setgifs(gifs_data);
        console.log("gifs...");
        console.log(gifs);
    }
    // getGifs();
    return (
        <>
            <h5>{category}</h5>
            <Divider></Divider>
            <Row>
                {
                    gifs.map((gif, ind) => (
                        <Col key={gif.id}>
                            <GifGridItem  {...gif} />
                        </Col>
                    ))
                }
            </Row>
            {/* <List className="mt-4" bordered>
                {
                    gifs.map((gif, ind) => (
                        <GifGridItem key={gif.id} {...gif} />

                    ))
                }
            </List> */}


            <h4>{count}</h4>
            <Button appearance="primary" onClick={() => setCount(count + 1)}>Rsuite</Button>
        </>
    );
}

GifGrid.propTypes = propTypes;
GifGrid.defaultProps = defaultProps;
// #endregion

export default GifGrid;