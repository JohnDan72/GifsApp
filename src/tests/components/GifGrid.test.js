import React from 'react';
import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');


describe('Pruebas GifGridComponent' , () => {
    const category = 'Goku';

    test('should render GifGrid component', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();
    }); 

    test('debe mostrar items cuando se cargan imÃ¡genes', () => {

        const gifs = [
            {id:'asd',url:'https://localhost/image.jpg',title:'asdasd',shortName: 'asdasd'},
            {id:'asd2',url:'https://localhost/image.jpg',title:'asdasd',shortName: 'asdasd'},
            {id:'asd3',url:'https://localhost/image.jpg',title:'asdasd',shortName: 'asdasd'},
        ]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category}/> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('TimeIcon').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(3);
        
    });
    
})