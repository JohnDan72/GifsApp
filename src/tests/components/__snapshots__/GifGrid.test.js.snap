// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridComponent debe mostrar items cuando se cargan im√°genes 1`] = `
<Fragment>
  <h5>
    Goku
  </h5>
  <Divider />
  <Row
    className="my-2 animate__animated animate__fadeIn"
  >
    <Col
      className="mt-3"
      key="asd"
      lg={6}
      md={8}
      xs={12}
    >
      <GifGridItem
        id="asd"
        shortName="asdasd"
        title="asdasd"
        url="https://localhost/image.jpg"
      />
    </Col>
    <Col
      className="mt-3"
      key="asd2"
      lg={6}
      md={8}
      xs={12}
    >
      <GifGridItem
        id="asd2"
        shortName="asdasd"
        title="asdasd"
        url="https://localhost/image.jpg"
      />
    </Col>
    <Col
      className="mt-3"
      key="asd3"
      lg={6}
      md={8}
      xs={12}
    >
      <GifGridItem
        id="asd3"
        shortName="asdasd"
        title="asdasd"
        url="https://localhost/image.jpg"
      />
    </Col>
  </Row>
</Fragment>
`;

exports[`Pruebas GifGridComponent should render GifGrid component 1`] = `
<Fragment>
  <h5>
    Goku
  </h5>
  <Divider />
  <FlexboxGrid
    justify="center"
  >
    <FlexboxGridItem>
      <Time
        spin={true}
        style={
          Object {
            "fontSize": "3rem",
          }
        }
      />
    </FlexboxGridItem>
  </FlexboxGrid>
  <Row
    className="my-2 animate__animated animate__fadeIn"
  />
</Fragment>
`;
